lib/puppet/pops/issues.rb

```diff
@@ -152,680 +154,681 @@ def self.hard_issue(issue_code, *args, &block)
-    "Not a valid hex number #{value}"
+    _("Not a valid hex number %{value}") % { value: value }
   end
 
   INVALID_OCTAL_NUMBER = hard_issue :INVALID_OCTAL_NUMBER, :value do
-    "Not a valid octal number #{value}"
+    _("Not a valid octal number %{value}") % { value: value }
   end
 
   INVALID_DECIMAL_NUMBER = hard_issue :INVALID_DECIMAL_NUMBER, :value do
-    "Not a valid decimal number #{value}"
+    _("Not a valid decimal number %{value}") % { value: value }
   end
 
   NO_INPUT_TO_LEXER = hard_issue :NO_INPUT_TO_LEXER do
-    "Internal Error: No string or file given to lexer to process."
+    _("Internal Error: No string or file given to lexer to process.")
   end
 
   UNRECOGNIZED_ESCAPE = issue :UNRECOGNIZED_ESCAPE, :ch do
-    "Unrecognized escape sequence '\\#{ch}'"
+    _("Unrecognized escape sequence '\\%{ch}'") % { ch: ch }
   end
 
   UNCLOSED_QUOTE = hard_issue :UNCLOSED_QUOTE, :after, :followed_by do
-    "Unclosed quote after #{after} followed by '#{followed_by}'"
+    _("Unclosed quote after %{after} followed by '%{followed_by}'") % { after: after, followed_by: followed_by }
   end
 
   UNCLOSED_MLCOMMENT = hard_issue :UNCLOSED_MLCOMMENT do
-    'Unclosed multiline comment'
+    _('Unclosed multiline comment')
   end
 
   EPP_INTERNAL_ERROR = hard_issue :EPP_INTERNAL_ERROR, :error do
-    "Internal error: #{error}"
+    _("Internal error: %{error}") % { error: error }
   end
 
   EPP_UNBALANCED_TAG = hard_issue :EPP_UNBALANCED_TAG do
-    'Unbalanced epp tag, reached <eof> without closing tag.'
+    _('Unbalanced epp tag, reached <eof> without closing tag.')
   end
 
   EPP_UNBALANCED_COMMENT = hard_issue :EPP_UNBALANCED_COMMENT do
-    'Reaching end after opening <%# without seeing %>'
+    _('Reaching end after opening <%# without seeing %>')
   end
 
   EPP_UNBALANCED_EXPRESSION = hard_issue :EPP_UNBALANCED_EXPRESSION do
-    'Unbalanced embedded expression - opening <% and reaching end of input'
+    _('Unbalanced embedded expression - opening <% and reaching end of input')
   end
 
   HEREDOC_UNCLOSED_PARENTHESIS = hard_issue :HEREDOC_UNCLOSED_PARENTHESIS, :followed_by do
-    "Unclosed parenthesis after '@(' followed by '#{followed_by}'"
+    _("Unclosed parenthesis after '@(' followed by '%{followed_by}'") % { followed_by: followed_by }
   end
 
   HEREDOC_WITHOUT_END_TAGGED_LINE = hard_issue :HEREDOC_WITHOUT_END_TAGGED_LINE do
-    'Heredoc without end-tagged line'
+    _('Heredoc without end-tagged line')
   end
 
   HEREDOC_INVALID_ESCAPE = hard_issue :HEREDOC_INVALID_ESCAPE, :actual do
-    "Invalid heredoc escape char. Only t, r, n, s,  u, L, $ allowed. Got '#{actual}'"
+    _("Invalid heredoc escape char. Only t, r, n, s,  u, L, $ allowed. Got '%{actual}'") % { actual: actual }
   end
 
   HEREDOC_INVALID_SYNTAX = hard_issue :HEREDOC_INVALID_SYNTAX do
-    'Invalid syntax in heredoc expected @(endtag[:syntax][/escapes])'
+    _('Invalid syntax in heredoc expected @(endtag[:syntax][/escapes])')
```

Comment:
@Magisus @MosesMendoza another instance of "Heredoc", should it remain in english?